dashboard/src/components/providers/AppProvider.tsx

```diff
@@ -446,36 +449,31 @@ const middlewareReducer = async (
       let updatedAutoCompleteRes: any;
       let runErr: any;
       try {
```

Comment:
Can this outer `try` even catch anything anymore? The only uncaught line is the `dispatch` line, which should be an infallible function call.

I think you can drop the outer try-catch-finally, deleting the catch block and always running the finally block.